src/main/kotlin/com/google/androidstudiopoet/input/NewInputExample.json

```diff
@@ -0,0 +1,103 @@
+{
+  //Project configuration is wrapped in another object to help us deal with versioning
+  "inputVersion": "0.0.1",
+  "projectConfig": {
+    "projectName": "genny",
+    "root": "./modules/",
+    "buildSystemConfig": {
+      //Optional param, but it is possible to specify what exactly build system setup should be used
+    },
+    "moduleConfigs": [
+      //Even though there are only 3 blocks here, in total 11 modules would be generated.
+      //One other thing, I believe that we should forbid dependencies withing one block, only interblock dependencies are allowed.
+      {
+        //Andorid Application Module
+        "moduleType": "ANDROID",
+        "moduleNamePrefix": "app",
+        "isApplication": true,
+
+        "numActivities": 8,
+        "resourcesConfig": {
+          "stringCount": 10,
+          "imageCount": 9,
+          "layoutCount": 7
+        },
+
+        "javaPackageCount": 3,
+        "javaClassCount": 8,
+        "javaMethodCount": 2000,
+
+        "kotlinPackageCount": 3,
+        "kotlinClassCount": 5,
+        "kotlinMethodCount": 1000,
+
+        "dependencies": [
+          {
+            //Depends on the first "pure" module
+            "type": "SINGLE",
+            "moduleNamePrefix": "pure"
+          },
+          {
+            //Depends on the all five "andoirdLib" modules
+            "type": "SINGLE",
+            "moduleNamePrefix": "androidLib",
+            "moduleCount": 5
+          }
+        ],
+        "flavours": [
```

Comment:
Would it be possible to also support input flavors as done in https://github.com/borisf/android-studio-poet/pull/31/commits/917229bf4eb72f1f3b57096ec47d43a96a26f1c5, that way helps when adding many flavors.

Reply:
The problem with that approach is that amount of flavour*dimens grows like n^2. We can think about adding "bulk" flavour/dimen generation like it is done with packages, would that work for you? Then there'll just be multiple types of FlavourConfig

Reply:
Yes, the number of variants is the product of the number of flavors per dimension, it grows really fast (exponentially assuming each dimension has the same number of flavors).

Adding bulk as in packages can also work (having a prefix, a dimension and a counter). Another way is by adding an optional counter, similar to what is done with the modules.

Reply:
Sorry, I meant modules, not packages. I'll think about it and push to this branch want I'll come up with.

Reply:
I think that having two counters instead of one is more practical, one for the number of flavours the other for dimensions, then it would be more flexible.

Reply:
Every dimension can have a different number of flavors. What the input would look like with what I said is something like this:

"flavours": [
  {
    "name": "red",
    "count": 5,
    "dimension": "color"
  },
  {
    "name": "green",
    "count": 2,
    "dimension": "color"
    "buildConfig": {
             "minSdkVersion": 21
    }
  },
  {
    "name": "paid",
    "dimension": "license"
  },
  {
    "name": "free",
    "dimension": "license"
  }
]

This would generate 7 flavors ("red0" to "red4", "green0" and "green1") for dimension "color" and 2 for "license" (the total number of variants in this case would be 28 ( 7 in color x 2 in license x {debug | release} )

Reply:
I think this the format is perfect. We can make it a tiny little bit more flexible. We can add one more param "dimensionCounter", then we can easily create tons of dimensions with tons of flavours. Otherwise we would be forced to write one block for each dimension